### PR TITLE
[Fix-16322] Set default registry-disconnect-strategy:strategy to stop

### LINKED ISSUE #16322: [Bug] Fix zk registry reconnect after session timeout cannot write heartbeat to zk
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

This might caused by we didn't recreate the zk client after reconnected, the simple way to fix this is change the registry-disconnect-strategy to stop and restart the service outside if the server cannot connect to zk.

### What you expected to happen

xx

### How to reproduce

xx

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close #16322 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-master/src/test/resources/application.yaml [hunk 1]
```diff
@@ -127,9 +127,7 @@ master:
   kill-application-when-task-failover: true
   registry-disconnect-strategy:
     # The disconnect strategy: stop, waiting
-    strategy: waiting
-    # The max waiting time to reconnect to registry if you set the strategy to waiting
-    max-waiting-time: 100s
+    strategy: stop
```
Comment (SbloodyS) [inline dolphinscheduler-master/src/test/resources/application.yaml:130]: Should we change alert-server and api-server?

Comment (ruanwenjun) [reply dolphinscheduler-master/src/test/resources/application.yaml:130]: Only master/worker have this config.

Comment (SbloodyS) [reply dolphinscheduler-master/src/test/resources/application.yaml:130]: Okay.

**Review by SbloodyS — APPROVED** [review]
LGTM

**Review by Gallardot — APPROVED** [review]
LGTM

**Pull request merged**